packages/ui/src/components/va-carousel/VaCarousel.stories.ts

```diff
@@ -1,12 +1,203 @@
+import { expect } from '@storybook/jest'
+import { sleep } from '../../utils/sleep'
+
+const carouselSetup = {
+  data () {
+    return {
+      currentSlide: 0,
+      areArrowsVisible: true,
+      areIndicatorsVisible: true,
+      triggerOnHover: true,
+      items: [
+        'https://picsum.photos/id/450/1500',
+        'https://picsum.photos/id/450/1501',
+        'https://picsum.photos/id/450/1502',
+        'https://picsum.photos/id/450/1503',
+        'https://picsum.photos/id/450/1504',
+      ],
+    }
+  },
+}
 
 export default {
   title: 'VaCarousel',
   component: VaCarousel,
+  tags: ['autodocs'],
+}
+
+export const Default = () => ({
+  components: { VaCarousel },
+  ...carouselSetup,
+  template: `
+    <VaCarousel :items="items" v-model="currentSlide" />
+  `,
+})
+
+Default.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const carousel = document.querySelectorAll('div.va-carousel__slide')
+  const slideTwo = canvas.getByRole('button', { name: 'go slide 2' }) as HTMLElement
+  const goForward = canvas.getByRole('button', { name: 'go next slide' }) as HTMLElement
+
+  await step('Should head to slide 2', async () => {
+    userEvent.click(slideTwo)
+    await sleep()
+    expect(carousel[1].getAttribute('aria-current')).toEqual('true')
+  })
+
+  await step('Should head to slide 3 by clicking on right arrow', async () => {
+    userEvent.click(goForward)
+    await sleep()
+    expect(carousel[2].getAttribute('aria-current')).toEqual('true')
+  })
+}
+
+export const Stateful = () => ({
+  components: { VaCarousel },
+  ...carouselSetup,
+  template: `
+    [true] - should slide
+    <VaCarousel :items="items" stateful />
+    [false] - should not slide
+    <VaCarousel :items="items" />
+  `,
+})
+
+Stateful.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const carousel = document.querySelectorAll('div.va-carousel__slide')
+  const [statefulslideTwo, slideTwo] = canvas.getAllByRole('button', { name: 'go slide 2' }) as HTMLElement[]
+
+  await step('Should head to slide 2', async () => {
+    userEvent.click(statefulslideTwo)
+    await sleep()
+    expect(carousel[1].getAttribute('aria-current')).toEqual('true')
+  })
+
+  await step('Should do nothing', async () => {
+    userEvent.click(slideTwo)
+    await sleep()
+    expect(carousel[6].getAttribute('aria-current')).toEqual('false')
+  })
+}
+
+export const Vertical = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <VaCarousel :items="items" v-model="currentSlide" vertical />
+  `,
+})
+
+export const Arrows = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <va-switch
+      v-model="areArrowsVisible"
+      class="mb-6"
+      label="Arrows"
+      size="small"
+    />
+    <VaCarousel :items="items" v-model="currentSlide" :arrows="areArrowsVisible" />
+  `,
+})
+
+export const Indicators = () => ({
```

Comment:
Let's split `indicators` and `indicator-trigger` props into different stories and handle them similar to `stateful`.

`indicator-trigger` might also need story for `none`?